skrf/calibration/deembedding.py

```diff
@@ -996,6 +996,14 @@ def __init__(self, dummy_2xthru, name=None,
             not enough point in time domain to determine length and impedance
             of midpoint. Parameter `verbose` could be used for diagnostic in
             ifft mode (default: False)
+            
+        forced_z0_line:
+            If a value is specified, manually force this value for the midpoint
+            impedance z0_line. The fixtures are renormalized with this value.
+            This is only usefull in the case where the midpoint impedance is
+            non-uniform and the x length ± 1 sample error caused by the delay

```

Comment:
what is "x length"? Should this be 2xthru length? Also I don't know what is meant by non-uniform midpoint impedance.

I understood that the method first uses s21 impulse response to determine the thru length, then it uses s11 step response at the half length to get the impedance at the split plane. Since each point in the time response corresponds to reflections from some finite length of line the method requires that the thru length is few time points long so that only the line reflections are in the mid time point. I don't think there is any requirement for length to be a multiple of time resolution. If that is a requirement, then fft should be zero padded or the output interpolated to get the length more accurately.

For very short thrus the time domain impedance determination doesn't work since there is no time point that has only the line reflections. In that case impedance could be given manually, but since e001 and e002 requires time domain transform at the midpoint the split thru includes some amount of launch.

Reply:
Hi @Ttl,


There's a history of discussion during my testing here with @mhuser prior to this pull request. 
Perhaps taking a look at this https://github.com/mhuser/scikit-rf/issues/2#issuecomment-1149356021 may help in setting the documentation wording better.

Reply:
Yeah, so it is meant for very short thrus where there aren't enough points in the time domain to determine the split plane impedance. Any choice of time point for the impedance determination includes also reflections from the launches. It should be noted that even if the impedance is given manually, some amount of reflections from opposite side launch is still included in the split thrus since there isn't enough time resolution to separate them. It does improve the result though and I'm not saying there is anything wrong with the feature, just that the documentation should be improved.

Reply:
I am rewriting the docstring in this sense.